### PR TITLE
Pypo permission fix

### PR DESCRIPTION
**Pull request by xabispacebiker**
Fix for issue https://github.com/LibreTime/libretime/issues/307

Additionally corrected some related bugs

### CODE REVIEW

File: airtime_mvc/application/common/HTTPHelper.php [hunk 1]
```diff
@@ -37,6 +37,9 @@ public static function getStationUrl($secured = true)
         if ($baseDir[0] != "/") {
             $baseDir = "/" . $baseDir;
         }
+        if ($baseDir[strlen($baseDir)-1] != "/") {
```
Comment (hairmare) [inline airtime_mvc/application/common/HTTPHelper.php]: You can use `substr($baseDir, -1)` to get the strings last char in a more straight forward fashion.

File: python_apps/pypo/pypo/pypofile.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def copy_file(self, media_item):
                         handle.write(chunk)
 
                 #make file world readable
-                os.chmod(dst, stat.S_IRUSR | stat.S_IRGRP | stat.S_IROTH)
+                os.chmod(dst, stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH)
```
Comment (hairmare) [inline python_apps/pypo/pypo/pypofile.py]: Awesome catch!

File: airtime_mvc/application/common/HTTPHelper.php (hunk 1 above)
Comment (xabispacebiker) [reply airtime_mvc/application/common/HTTPHelper.php]: Great, I updated the code.. I guess not but.. should I send the request again?

**Comment by hairmare** [comment]
Thanks for the patch!

**Comment by xabispacebiker** [comment]
@hairmare , sorry but I checked the merge and realized there are missing changes which are very important. I am running right now with this pypo file and it seems to resolve all my issues with liquidsoap + it does not produce so many logs as it did before, which is a good point too. 

Mostly it was happening because I run libretime over the port https at 443.

I guess it was a fault from my side, as I made my head a mess with netbean's github thing... anyway, I have updated the full pypofile.py with the corrected version: https://github.com/xabispacebiker/libretime/blob/pypo-permission-fix/python_apps/pypo/pypo/pypofile.py

My apologize

**Comment by hairmare** [comment]
[These](https://github.com/LibreTime/libretime/compare/master...xabispacebiker:pypo-permission-fix) seem to make sense to me. There are some small whitespace issues that might make sense to clean up (not all line are aligned properly on the left side). Could you address those and then make a new PR from the branch? I'd like to get these changes merged before I tag 3.0.0-alpha.3.

**Pull request merged**